packages/nx/src/migrations/update-16-8-0/escape-dollar-sign-env-variables.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { createTreeWithEmptyWorkspace } from '../../generators/testing-utils/create-tree-with-empty-workspace';
+import { addProjectConfiguration } from '../../generators/utils/project-configuration';
+import escapeDollarSignEnvVariables from './escape-dollar-sign-env-variables';
+
+describe('escape $ in env variables', () => {
+  let tree;
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace();
+  });
+
+  it('should escape $ in env variables in .env file', () => {
+    tree.write(
+      '.env',
+      `dollar=$
+NX_SOME_VAR=$ABC`
+    );
+    escapeDollarSignEnvVariables(tree);
+    expect(tree.read('.env', 'utf-8')).toEqual(`dollar=\$
+NX_SOME_VAR=\$ABC`);
```

Comment:
You need to escape the `\` in all of these checks.

```suggestion
    expect(tree.read('.env', 'utf-8')).toEqual(`dollar=\\$
NX_SOME_VAR=\\$ABC`);
```

Reply:
changed